### PR TITLE
App Pagination to Data

### LINKED ISSUE #22: Add pagination data request (optional)
When an API request includes a: page_offset or page_size GET parameter use the optional paging control and output.

Info:
- page[*] *required for pagination, must specify either offset or size
- page[size] = 25 *default
- page[offset] = 1 *default

Output will include a meta.page property that contains the paging information.
```
{
  "meta": {
    "page": {
      "size": 25,
      "offset": 2,
      "total_records": 80
      "total_pages": 4
    }
  },
  "data": {... ...}
}
```

Use: https://github.com/amatsuda/kaminari, to do the data request paging.

### PR DESCRIPTION
**Pull request by chadzink**
Fixes #22

### CODE REVIEW

**Comment by jweakley** [comment]
Looks like the travis CI setting are incorrect. We need to remove the unused versions of Ruby from the build options.

File: lib/api_me/pagination.rb [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    def results
+      if is_paging
+        page.per.scope
+      else
+        scope
+      end
+    end
+
+    def page_meta
+      unless is_paging
```
Comment (jweakley) [inline lib/api_me/pagination.rb]: You can rewrite these 3 lines as:
`return Hash.new unless is_paging?`

File: lib/api_me/pagination.rb [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      self
+    end
+
+    private
+
+    def default_page_size
+      Kaminari.config.default_per_page
+    end
+
+    def is_paging
```
Comment (jweakley) [inline lib/api_me/pagination.rb]: Rename this to `is_paging?`

File: lib/api_me/pagination.rb [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+      end
+    end
+
+    def page_meta
+      unless is_paging
+        return {}
+      end
+
+      {
+        size: self.page_size.nil? ? default_page_size : self.page_size,
```
Comment (jweakley) [inline lib/api_me/pagination.rb]: Unneeded `self` in both spots.

File: lib/api_me/pagination.rb [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        offset: self.page_offset,
```
Comment (jweakley) [inline lib/api_me/pagination.rb]: Unneeded `self

File: lib/api_me/pagination.rb (hunk 2 above)
Comment (jweakley) [inline lib/api_me/pagination.rb]: You could also call this `paging?`

File: lib/api_me/pagination.rb [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+        record_count: self.scope.size,
```
Comment (jweakley) [inline lib/api_me/pagination.rb]: Unneeded `self`

File: lib/api_me/pagination.rb [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+        total_records: self.scope.total_count,
```
Comment (jweakley) [inline lib/api_me/pagination.rb]: Unneeded `self`

File: lib/api_me/pagination.rb [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+        total_pages: self.scope.total_pages
```
Comment (jweakley) [inline lib/api_me/pagination.rb]: Unneeded `self`

**Comment by jweakley** [comment]
:+1: once those issues are addressed.

**Comment by chadzink** [comment]
I adjusted the ruby source with recommendations
I am not sure how to fix the travis CI issues or what that means really. @jweakley maybe we can discuss when you get time?

**Comment by jweakley** [comment]
@chadzink, sounds like a plan. I'll walk you through it when you get into the office.

**Pull request merged**